orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -134,13 +134,18 @@ def forward(ctx, *inputs, **kwargs):
                     Module outputs are returned to the user
                     '''
 
+                    # Disable materializing grads then None object will not be converted to a tensor filled with zeros prior to calling backward.
+                    ctx.set_materialize_grads(False)
+
                     # Use IO binding
                     _create_iobinding(self._training_io_binding, inputs, self._onnx_training, self._device)
 
                     # Run and return module outputs.
                     forward_outputs, run_id = self._training_session.run_forward(self._training_io_binding, self._run_options)
                     user_outputs = tuple(_ort_output_to_torch_tensor(forward_output) for forward_output in forward_outputs)
                     ctx.run_id = run_id
+                    # Save shape, device and type info for materializing tensor in backward if output grad is None.
```

Comment:
nit. move to the line after "ctx.set_materialize_grads(False)", since they are connected.

Reply:
user_outputs are generated after run_forward. Let me move the set_materialize_grads here.